### PR TITLE
Run rustfmt on Travis.

### PR DESCRIPTION
**Pull request by snim2**
Also reformats source code.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,2 +1,10 @@
 language: rust
 rust: nightly
+cache: cargo
+before_script:
+- export PATH="$PATH:$HOME/.cargo/bin"
+- which rustfmt || cargo install rustfmt
```
Comment (vext01) [inline .travis.yml]: is rustfmt in the apt repo?

File: .travis.yml [hunk 2]
```diff
@@ -1,2 +1,10 @@
...
+script:
+- cargo fmt -- --write-mode=diff
```
Comment (vext01) [inline .travis.yml]: So this fails is the code is not in the canonical format?

File: grammars/java.y [hunk 3]
```diff
@@ -192,44 +192,44 @@ modifier :    "PUBLIC" | "PROTECTED" | "PRIVATE"
...
     ;
 
-class_declaration : 
+class_declaration :
     modifiers_opt "CLASS" "IDENTIFIER" type_parameters_opt
       super_opt interfaces_opt class_body ;
 
 super :    "EXTENDS" class_type;
 
-super_opt :
```
Comment (vext01) [inline grammars/java.y]: I approve of the removal of trailing spaces!

File: .travis.yml (hunk 2 above)
Comment (snim2) [inline .travis.yml]: Yes (I'm not sure whether this is desirable yet, I'm just copying the advice here: https://github.com/rust-lang-nursery/rustfmt)

File: .travis.yml (hunk 1 above)
Comment (snim2) [inline .travis.yml]: Doesn't seem to be (on my laptop)

Comment (vext01) [inline .travis.yml]: ok

File: .travis.yml (hunk 2 above)
Comment (vext01) [inline .travis.yml]: That's fine. The alternative would be to use a post-commit hook to re-format the code in-situ, but I think I prefer how you've done it here.

**Comment by snim2** [comment]
Yes please!

**Pull request merged**